src/DataTypes/object.tsx

```diff
@@ -0,0 +1,400 @@
+			  }
+			: (() => {
+					const keyLabel = getKeyLabel(s);
+					const valueLabel = getValueLabel(s);
+					return {
+						key_contains: `${keyLabel} contains`,
+						key_not_contains: `${keyLabel} does not contain`,
+						key_is: `${keyLabel} is`,
+						key_matches_re: `${keyLabel} matches RegEx`,
+						key_not_matches_re: `${keyLabel} does not match RegEx`,
+						value_is: `${valueLabel} is`,
+						value_contains: `${valueLabel} contains`,
+						value_not_contains: `${valueLabel} does not contain`,
+						value_matches_re: `${valueLabel} matches RegEx`,
+						value_not_matches_re: `${valueLabel} does not match RegEx`,
+					};
+			  })()),
+	};
+};
+
+const keySpecificOperators = [
+	'key_is',
+	'key_contains',
+	'key_not_contains',
+	'key_matches_re',
+	'key_not_matches_re',
+];
+
+const valueSpecificOperators = [
+	'value_is',
+	'value_contains',
+	'value_not_contains',
+	'value_matches_re',
+	'value_not_matches_re',
+];
+
+const getValueForOperation = (
+	operator: string,
+	schema: JSONSchema,
+	value: string | object,
+) => {
+	// Determine if the operation is key-specific or value-specific
+	const isKeyOperation = keySpecificOperators.includes(operator);
+	const isValueOperation = valueSpecificOperators.includes(operator);
+
+	// Find the schema key or value based on the operation type
+	const schemaField = isKeyOperation
+		? 'key'
+		: isValueOperation
+		? 'value'
+		: null;
+	const schemaProperty = schemaField
+		? findKey(schema.properties!, { description: schemaField })
+		: null;
+
+	// Return the appropriate value format based on the operation type
+	return schemaProperty
+		? typeof value === 'string'
+			? { [schemaProperty]: value }
+			: pick(value, schemaProperty)
+		: value;
+};
+
+const getTitleForOperation = (
+	operator: OperatorSlug,
+	schema: JSONSchema,
+	value: string | object,
+) => {
+	if (typeof value !== 'object' || !schema.properties) {
+		return schema.title;
+	}
+
+	// Combine key and value specific operators for easier checking
+	const allSpecificOperators = [
+		...keySpecificOperators,
+		...valueSpecificOperators,
+	];
+
+	// Extract the first key from the value object
+	const firstKeyOfValue = Object.keys(value)[0];
+
+	// Proceed only if the operator is in the list of specific operators and the property exists
+	if (
+		allSpecificOperators.includes(operator) &&
+		schema.properties[firstKeyOfValue]
+	) {
+		const property = schema.properties[firstKeyOfValue];
+
+		// Ensure the property is an object and has a title
+		if (typeof property === 'object' && property.title) {
+			return property.title;
+		}
+	}
+
+	// Default return if none of the conditions above are met
+	return schema.title;
+};
+
+export type OperatorSlug =
+	| KeysOfUnion<ReturnType<typeof operators>>
+	| typeof FULL_TEXT_SLUG;
+
+export const createFilter: CreateFilter<OperatorSlug> = (
+	field,
+	operator,
+	value,
+	schema,
+) => {
+	if (!schema) {
+		return {};
+	}
+
+	const internalValue = !!isKeyValueObj(schema)
```

Comment:
move the `isKeyValueObj` call to a const so that you don't have to call it twice?